proof/crefine/ARM/Arch_C.thy

```diff
@@ -1885,6 +1883,132 @@ lemma performPageInvocationMapPDE_ccorres:
   apply (auto simp: unat_arith_simps unat_word_ariths)
   done
 
+lemma length_of_msgRegisters:
+  "length ARM_H.msgRegisters = 4"
+  by (auto simp: msgRegisters_unfold)
```

Comment:
We usually try to avoid expanding magic numbers in the proofs so that we don't have to update the proofs when such constants change. I think @Xaphiosis had a similar lemma in the fastpath rearrangement PR recently that achieved the same but stayed schematic. If you rebase over master this lemma might already be in scope at this point.

Reply:
`length_msgRegisters` in `ArchMove_C` on every arch.

Reply:
Will have a look, this was copied from ARM_HYP so I'll try to fix it there too